### ISSUE TITLE
SUSE Leap 42.2: salt-jenkins packages don't install properly.

### ISSUE THREAD

**Issue by rallytime** [saltstack/salt-jenkins#361]
After updating the image from Linode in issue #358, the salt-jenkins states are no longer running properly after a basic `state.sls git.salt` call:
```
18:39:49 local:
18:39:49 ----------
18:39:49           ID: force-sync-all
18:39:49     Function: module.run
18:39:49         Name: saltutil.sync_all
18:39:49       Result: True
18:39:49      Comment: Module function saltutil.sync_all executed
18:39:49      Started: 18:38:56.467943
18:39:49     Duration: 5532.107 ms
18:39:49      Changes:   
18:39:49               ----------
18:39:49               ret:
18:39:49                   ----------
18:39:49                   beacons:
18:39:49                   engines:
18:39:49                   grains:
18:39:49                       - grains.external_ip
18:39:49                       - grains.python_executable
18:39:49                   log_handlers:
18:39:49                   modules:
18:39:49                       - modules.pip
18:39:49                       - modules.runtests
18:39:49                       - modules.six
18:39:49                   output:
18:39:49                   proxymodules:
18:39:49                   renderers:
18:39:49                   returners:
18:39:49                   sdb:
18:39:49                   states:
18:39:49                       - states.pip2_state
18:39:49                       - states.pip3_state
18:39:49                       - states.pip_state
18:39:49                       - states.runtests
18:39:49                   utils:
18:39:49 ----------
18:39:49           ID: curl
18:39:49     Function: pkg.installed
18:39:49       Result: False
18:39:49      Comment: An error was encountered while installing package(s): Zypper command failure: Package 'nodejs' not found.
18:39:49      Started: 18:39:02.644833
18:39:49     Duration: 2858.911 ms
18:39:49      Changes:   
18:39:49 ----------
18:39:49           ID: pip-install
18:39:49     Function: cmd.run
18:39:49         Name: curl -L 'https://bootstrap.pypa.io/get-pip.py' -o get-pip.py && python get-pip.py  -U pip
18:39:49       Result: False
18:39:49      Comment: One or more requisite failed: curl.curl
18:39:49      Started: 
18:39:49     Duration: 
18:39:49      Changes:   
18:39:49 ----------
18:39:49           ID: upgrade-installed-pip
18:39:49     Function: pip.installed
18:39:49         Name: pip
18:39:49       Result: False
18:39:49      Comment: One or more requisite failed: python.pip.pip-install
18:39:49      Started: 
18:39:49     Duration: 
18:39:49      Changes:   
18:39:49 ----------
18:39:49           ID: docker_py
18:39:49     Function: pip.installed
18:39:49         Name: docker
18:39:49       Result: False
18:39:49      Comment: One or more requisite failed: python.pip.pip-install
18:39:49      Started: 
18:39:49     Duration: 
18:39:49      Changes:   
18:39:49 ----------
18:39:49           ID: us_locale
18:39:49     Function: locale.present
18:39:49         Name: en_US.UTF-8
18:39:49       Result: True
18:39:49      Comment: Locale en_US.UTF-8 is already present
18:39:49      Started: 18:39:05.508920
18:39:49     Duration: 116.282 ms
18:39:49      Changes:   
18:39:49 ----------
18:39:49           ID: default_locale
18:39:49     Function: locale.system
18:39:49         Name: en_US.UTF-8
18:39:49       Result: True
18:39:49      Comment: Set system locale en_US.UTF-8
18:39:50      Started: 18:39:05.626203Connection to 173.255.255.127 closed.
18:39:50 
18:39:50     Duration: 53.041 ms
18:39:50      Changes:   
18:39:50               ----------
18:39:50               locale:
18:39:50                   en_US.UTF-8
18:39:50 ----------
18:39:50           ID: git
18:39:50     Function: pkg.installed
18:39:50       Result: True
18:39:50      Comment: Package git is already installed
18:39:50      Started: 18:39:05.679621
18:39:50     Duration: 9.038 ms
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: patch
18:39:50     Function: pkg.installed
18:39:50       Result: True
18:39:50      Comment: Package patch is already installed
18:39:50      Started: 18:39:05.688931
18:39:50     Duration: 8.889 ms
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: sed
18:39:50     Function: pkg.installed
18:39:50       Result: True
18:39:50      Comment: Package sed is already installed
18:39:50      Started: 18:39:05.698101
18:39:50     Duration: 9.183 ms
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: virtualenv
18:39:50     Function: pip.installed
18:39:50         Name: virtualenv==1.10
18:39:50       Result: False
18:39:50      Comment: One or more requisite failed: python.pip.pip-install
18:39:50      Started: 
18:39:50     Duration: 
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: certifi
18:39:50     Function: pip.installed
18:39:50       Result: False
18:39:50      Comment: One or more requisite failed: python.pip.pip-install
18:39:50      Started: 
18:39:50     Duration: 
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: mock
18:39:50     Function: pip.installed
18:39:50         Name: mock < 1.1.0
18:39:50       Result: False
18:39:50      Comment: One or more requisite failed: python.pip.pip-install
18:39:50      Started: 
18:39:50     Duration: 
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: six
18:39:50     Function: pip.installed
18:39:50       Result: False
18:39:50      Comment: One or more requisite failed: python.pip.pip-install
18:39:50      Started: 
18:39:50     Duration: 
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: gcc
18:39:50     Function: pkg.installed
18:39:50       Result: True
18:39:50      Comment: Package gcc is already installed
18:39:50      Started: 18:39:05.709054
18:39:50     Duration: 8.654 ms
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: python-dev
18:39:50     Function: pkg.installed
18:39:50         Name: python-devel
18:39:50       Result: True
18:39:50      Comment: Package python-devel is already installed
18:39:50      Started: 18:39:05.718012
18:39:50     Duration: 8.427 ms
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: timelib
18:39:50     Function: pip.installed
18:39:50       Result: False
18:39:50      Comment: One or more requisite failed: python.pip.pip-install
18:39:50      Started: 
18:39:50     Duration: 
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: coverage
18:39:50     Function: pip.installed
18:39:50         Name: coverage==3.7.1
18:39:50       Result: False
18:39:50      Comment: One or more requisite failed: python.pip.pip-install
18:39:50      Started: 
18:39:50     Duration: 
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: unittest-xml-reporting
18:39:50     Function: pip.installed
18:39:50       Result: False
18:39:50      Comment: One or more requisite failed: python.pip.pip-install
18:39:50      Started: 
18:39:50     Duration: 
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: apache-libcloud
18:39:50     Function: pip.installed
18:39:50         Name: apache-libcloud==1.0.0
18:39:50       Result: False
18:39:50      Comment: One or more requisite failed: python.pip.pip-install
18:39:50      Started: 
18:39:50     Duration: 
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: requests
18:39:50     Function: pip.installed
18:39:50       Result: False
18:39:50      Comment: One or more requisite failed: python.pip.pip-install
18:39:50      Started: 
18:39:50     Duration: 
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: keyring
18:39:50     Function: pip.installed
18:39:50         Name: keyring==5.7.1
18:39:50       Result: False
18:39:50      Comment: One or more requisite failed: python.pip.pip-install
18:39:50      Started: 
18:39:50     Duration: 
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: gnupg
18:39:50     Function: pip.installed
18:39:50         Name: python-gnupg
18:39:50       Result: False
18:39:50      Comment: One or more requisite failed: python.pip.pip-install
18:39:50      Started: 
18:39:50     Duration: 
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: cherrypy
18:39:50     Function: pip.installed
18:39:50       Result: False
18:39:50      Comment: One or more requisite failed: python.pip.pip-install
18:39:50      Started: 
18:39:50     Duration: 
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: python-etcd
18:39:50     Function: pip.installed
18:39:50         Name: python-etcd==0.4.2
18:39:50       Result: False
18:39:50      Comment: One or more requisite failed: python.pip.pip-install
18:39:50      Started: 
18:39:50     Duration: 
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: gitpython
18:39:50     Function: pip.installed
18:39:50         Name: GitPython
18:39:50       Result: False
18:39:50      Comment: One or more requisite failed: python.pip.pip-install
18:39:50      Started: 
18:39:50     Duration: 
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: supervisor
18:39:50     Function: pip2.installed
18:39:50       Result: False
18:39:50      Comment: One or more requisite failed: python.pip.pip-install
18:39:50      Started: 
18:39:50     Duration: 
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: boto
18:39:50     Function: pip.installed
18:39:50         Name: boto >= 2.33.0
18:39:50       Result: False
18:39:50      Comment: One or more requisite failed: python.pip.pip-install
18:39:50      Started: 
18:39:50     Duration: 
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: boto3
18:39:50     Function: pip.installed
18:39:50       Result: False
18:39:50      Comment: One or more requisite failed: python.pip.pip-install
18:39:50      Started: 
18:39:50     Duration: 
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: moto
18:39:50     Function: pip.installed
18:39:50         Name: moto == 0.4.31
18:39:50       Result: False
18:39:50      Comment: One or more requisite failed: python.pip.pip-install
18:39:50      Started: 
18:39:50     Duration: 
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: psutil
18:39:50     Function: pip.installed
18:39:50       Result: False
18:39:50      Comment: One or more requisite failed: python.pip.pip-install
18:39:50      Started: 
18:39:50     Duration: 
18:39:50      Changes:   
18:39:50 ----------
18:39:50           ID: tornado
18:39:50     Function: pip.installed
18:39:51         Name: tornado>=4.2.1,<4.5.0
18:39:51       Result: False
18:39:51      Comment: One or more requisite failed: python.pip.pip-install
18:39:51      Started: 
18:39:51     Duration: 
18:39:51      Changes:   
18:39:51 ----------
18:39:51           ID: pyvmomi
18:39:51     Function: pip.installed
18:39:51       Result: False
18:39:51      Comment: One or more requisite failed: python.pip.pip-install
18:39:51      Started: 
18:39:51     Duration: 
18:39:51      Changes:   
18:39:51 ----------
18:39:51           ID: pycrypto
18:39:51     Function: pip.installed
18:39:51         Name: pycrypto >= 2.6.1
18:39:51       Result: False
18:39:51      Comment: One or more requisite failed: python.pip.pip-install
18:39:51      Started: 
18:39:51     Duration: 
18:39:51      Changes:   
18:39:51 ----------
18:39:51           ID: setproctitle
18:39:51     Function: pip.installed
18:39:51       Result: False
18:39:51      Comment: One or more requisite failed: python.pip.pip-install
18:39:51      Started: 
18:39:51     Duration: 
18:39:51      Changes:   
18:39:51 ----------
18:39:51           ID: pyinotify
18:39:51     Function: pip.installed
18:39:51       Result: False
18:39:51      Comment: One or more requisite failed: python.pip.pip-install
18:39:51      Started: 
18:39:51     Duration: 
18:39:51      Changes:   
18:39:51 ----------
18:39:51           ID: msgpack-python
18:39:51     Function: pip.installed
18:39:51         Name: msgpack-python >= 0.4.2
18:39:51       Result: False
18:39:51      Comment: One or more requisite failed: python.pip.pip-install
18:39:51      Started: 
18:39:51     Duration: 
18:39:51      Changes:   
18:39:51 ----------
18:39:51           ID: jsonschema
18:39:51     Function: pip.installed
18:39:51       Result: False
18:39:51      Comment: One or more requisite failed: python.pip.pip-install
18:39:51      Started: 
18:39:51     Duration: 
18:39:51      Changes:   
18:39:51 ----------
18:39:51           ID: rfc3987
18:39:51     Function: pip.installed
18:39:51       Result: False
18:39:51      Comment: One or more requisite failed: python.pip.pip-install
18:39:51      Started: 
18:39:51     Duration: 
18:39:51      Changes:   
18:39:51 ----------
18:39:51           ID: strict_rfc3339
18:39:51     Function: pip.installed
18:39:51       Result: False
18:39:51      Comment: One or more requisite failed: python.pip.pip-install
18:39:51      Started: 
18:39:51     Duration: 
18:39:51      Changes:   
18:39:51 ----------
18:39:51           ID: pyopenssl
18:39:51     Function: pkg.installed
18:39:51         Name: python-pyOpenSSL
18:39:51       Result: True
18:39:51      Comment: Package python-pyOpenSSL is already installed
18:39:51      Started: 18:39:05.900614
18:39:51     Duration: 9.712 ms
18:39:51      Changes:   
18:39:51 ----------
18:39:51           ID: install_ruby
18:39:51     Function: pkg.installed
18:39:51         Name: ruby
18:39:51       Result: True
18:39:51      Comment: Package ruby is already installed
18:39:51      Started: 18:39:05.910706
18:39:51     Duration: 8.391 ms
18:39:51      Changes:   
18:39:51 ----------
18:39:51           ID: futures
18:39:51     Function: pip.installed
18:39:51       Result: False
18:39:51      Comment: One or more requisite failed: python.pip.pip-install
18:39:51      Started: 
18:39:51     Duration: 
18:39:51      Changes:   
18:39:51 ----------
18:39:51           ID: dnsutils
18:39:51     Function: pkg.installed
18:39:51         Name: bind-utils
18:39:51       Result: True
18:39:51      Comment: Package bind-utils is already installed
18:39:51      Started: 18:39:05.919798
18:39:51     Duration: 8.641 ms
18:39:51      Changes:   
18:39:51 ----------
18:39:51           ID: ioflo
18:39:51     Function: pip.installed
18:39:51       Result: False
18:39:51      Comment: One or more requisite failed: python.pip.pip-install
18:39:51      Started: 
18:39:51     Duration: 
18:39:51      Changes:   
18:39:51 ----------
18:39:51           ID: mysqldb
18:39:51     Function: pkg.installed
18:39:51         Name: python-MySQL-python
18:39:51       Result: True
18:39:51      Comment: The following packages were installed/updated: python-MySQL-python
18:39:51      Started: 18:39:05.929105
18:39:51     Duration: 9818.255 ms
18:39:51      Changes:   
18:39:51               ----------
18:39:51               libmysqlclient18:
18:39:51                   ----------
18:39:51                   new:
18:39:51                       10.0.29-18.1
18:39:51                   old:
18:39:51               mariadb:
18:39:51                   ----------
18:39:51                   new:
18:39:51                       10.0.29-18.1
18:39:51                   old:
18:39:51               mariadb-client:
18:39:51                   ----------
18:39:51                   new:
18:39:51                       10.0.29-18.1
18:39:51                   old:
18:39:51               mariadb-errormessages:
18:39:51                   ----------
18:39:51                   new:
18:39:51                       10.0.29-18.1
18:39:51                   old:
18:39:51               python-MySQL-python:
18:39:51                   ----------
18:39:51                   new:
18:39:51                       1.2.5-8.3
18:39:51                   old:
18:39:51 ----------
18:39:51           ID: dnspython
18:39:51     Function: pip.installed
18:39:51       Result: False
18:39:51      Comment: One or more requisite failed: python.pip.pip-install
18:39:51      Started: 
18:39:51     Duration: 
18:39:51      Changes:   
18:39:51 ----------
18:39:51           ID: npm
18:39:51     Function: pkg.installed
18:39:51         Name: nodejs
18:39:51       Result: False
18:39:51      Comment: An error was encountered while installing package(s): Zypper command failure: Package 'nodejs' not found.
18:39:51      Started: 18:39:16.112921
18:39:51     Duration: 1304.839 ms
18:39:51      Changes:   
18:39:51 ----------
18:39:51           ID: bower
18:39:51     Function: npm.installed
18:39:51       Result: False
18:39:51      Comment: One or more requisite failed: npm.npm
18:39:51      Started: 
18:39:51     Duration: 
18:39:51      Changes:   
18:39:51 ----------
18:39:51           ID: create-swap-file
18:39:51     Function: cmd.run
18:39:51         Name: dd if=/dev/zero of=/.salt-runtests.swapfile bs=2048 count=1M
18:39:51       Result: True
18:39:51      Comment: Command "dd if=/dev/zero of=/.salt-runtests.swapfile bs=2048 count=1M" run
18:39:51      Started: 18:39:17.552891
18:39:51     Duration: 27507.374 ms
18:39:51      Changes:   
18:39:51               ----------
18:39:51               pid:
18:39:51                   20960
18:39:51               retcode:
18:39:51                   0
18:39:51               stderr:
18:39:51                   1048576+0 records in
18:39:51                   1048576+0 records out
18:39:51                   2147483648 bytes (2.1 GB, 2.0 GiB) copied, 27.4836 s, 78.1 MB/s
18:39:51               stdout:
18:39:51 ----------
18:39:51           ID: make-swap
18:39:51     Function: cmd.run
18:39:51         Name: mkswap /.salt-runtests.swapfile
18:39:51       Result: True
18:39:51      Comment: Command "mkswap /.salt-runtests.swapfile" run
18:39:52      Started: 18:39:45.061546
18:39:52     Duration: 552.827 ms
18:39:52      Changes:   
18:39:52               ----------
18:39:52               pid:
18:39:52                   21933
18:39:52               retcode:
18:39:52                   0
18:39:52               stderr:
18:39:52                   mkswap: /.salt-runtests.swapfile: insecure permissions 0644, 0600 suggested.
18:39:52               stdout:
18:39:52                   Setting up swapspace version 1, size = 2 GiB (2147479552 bytes)
18:39:52                   no label, UUID=1af0c7f1-1e2d-41c7-a721-4566af27a6fe
18:39:52 ----------
18:39:52           ID: add-extra-swap
18:39:52     Function: mount.swap
18:39:52         Name: /.salt-runtests.swapfile
18:39:52       Result: True
18:39:52      Comment: Swap /.salt-runtests.swapfile activated
18:39:52      Started: 18:39:45.615782
18:39:52     Duration: 80.64 ms
18:39:52      Changes:   
18:39:52               ----------
18:39:52               priority:
18:39:52                   -2
18:39:52               size:
18:39:52                   2097148
18:39:52               type:
18:39:52                   file
18:39:52               used:
18:39:52                   0
18:39:52 ----------
18:39:52           ID: install-dmidecode
18:39:52     Function: pkg.installed
18:39:52         Name: dmidecode
18:39:52       Result: True
18:39:52      Comment: The following packages were installed/updated: dmidecode
18:39:52      Started: 18:39:45.696770
18:39:52     Duration: 3750.989 ms
18:39:52      Changes:   
18:39:52               ----------
18:39:52               dmidecode:
18:39:52                   ----------
18:39:52                   new:
18:39:52                       3.0-2.3
18:39:52                   old:
18:39:52 ----------
18:39:52           ID: SaltTesting
18:39:52     Function: pip.installed
18:39:52         Name: salttesting==2016.9.7
18:39:52       Result: False
18:39:52      Comment: One or more requisite failed: python.pip.pip-install
18:39:52      Started: 
18:39:52     Duration: 
18:39:52      Changes:   
18:39:52 ----------
18:39:52           ID: pytest
18:39:52     Function: pip.installed
18:39:52       Result: False
18:39:52      Comment: One or more requisite failed: python.pip.pip-install
18:39:52      Started: 
18:39:52     Duration: 
18:39:52      Changes:   
18:39:52 ----------
18:39:52           ID: pytest-tempdir
18:39:52     Function: pip.installed
18:39:52       Result: False
18:39:52      Comment: One or more requisite failed: python.pip.pip-install
18:39:52      Started: 
18:39:52     Duration: 
18:39:52      Changes:   
18:39:52 ----------
18:39:52           ID: pytest-catchlog
18:39:52     Function: pip.installed
18:39:52         Name: git+https://github.com/eisensheng/pytest-catchlog.git@develop#egg=Pytest-catchlog
18:39:52       Result: False
18:39:52      Comment: One or more requisite failed: python.pip.pip-install
18:39:52      Started: 
18:39:52     Duration: 
18:39:52      Changes:   
18:39:52 ----------
18:39:52           ID: pytest-helpers-namespace
18:39:52     Function: pip.installed
18:39:52       Result: False
18:39:52      Comment: One or more requisite failed: python.pip.pip-install
18:39:52      Started: 
18:39:52     Duration: 
18:39:52      Changes:   
18:39:52 ----------
18:39:52           ID: pytest-salt
18:39:52     Function: pip.installed
18:39:52         Name: git+https://github.com/saltstack/pytest-salt.git@master#egg=pytest-salt
18:39:52       Result: False
18:39:52      Comment: One or more requisite failed: python.pip.pip-install
18:39:52      Started: 
18:39:52     Duration: 
18:39:52      Changes:   
18:39:52 ----------
18:39:52           ID: /testing
18:39:52     Function: file.directory
18:39:52       Result: True
18:39:52      Comment: Directory /testing updated
18:39:52      Started: 18:39:49.466443
18:39:52     Duration: 7.497 ms
18:39:52      Changes:   
18:39:52               ----------
18:39:52               /testing:
18:39:52                   New Dir
18:39:52 ----------
18:39:52           ID: clone-salt-repo
18:39:52     Function: git.latest
18:39:52         Name: https://github.com/saltstack/salt.git
18:39:52       Result: False
18:39:52      Comment: One or more requisite failed: python.pyinotify.pyinotify, python.tornado.tornado, bower.bower, python.gitpython.gitpython, python.unittest-xml-reporting.unittest-xml-reporting, python.docker.docker_py, python.gnupg.gnupg, python.requests.requests, python.libcloud.apache-libcloud, python.etcd.python-etcd, python.psutil.psutil, python.pyvmomi.pyvmomi, python.moto.moto, python.certifi.certifi, python.boto.boto, python.salttesting.SaltTesting, python.coverage.coverage, npm.npm, python.supervisor.supervisor, python.futures.futures, python.virtualenv.virtualenv, python.keyring.keyring, python.mock.mock, python.dns.dnspython, python.timelib.timelib, python.pycrypto.pycrypto, python.ioflo.ioflo, python.cherrypy.cherrypy
18:39:52      Started: 
18:39:52     Duration: 
18:39:52      Changes:   
18:39:52 
18:39:52 Summary for local
18:39:52 -------------
18:39:52 Succeeded: 17 (changed=8)
18:39:52 Failed:    44
18:39:52 -------------
18:39:52 Total states run:     61
```

@dubb-b FYI - I am not sure if this has something to do with the mirror or not. I suspect not since the `pkg.installed` calls are also failing.

**Comment by rallytime** [comment]
Oh, and this is failing for all branches in `salt-jenkins`.

**Comment by dubb-b** [comment]
@rallytime No, this is due to the Zypper packages not being available. I would say that it could be connectivity to the Zypper repo, not sure though. Also it could be that they changed the `node` package name.

Ill look at it real quick.

**Comment by rallytime** [comment]
Fixed with #362
